From: Peter Rashleigh <prashleigh@questertangent.com>
Date: Tue, 8 Oct 2024 14:33:52 -0700
Subject: [PATCH] Fix uninitialized session pointers

If crypto_copy_hash_state is called with an invalid sid, then
crypto_get_sessions_by_id can generate a kernel oops when it tries to
use the session pointers. Initializing them to NULL avoids the issue.
---
 ioctl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ioctl.c b/ioctl.c
index 747d32d5da66..db35200a42bb 100644
--- a/ioctl.c
+++ b/ioctl.c
@@ -506,7 +506,8 @@ out:
 static int
 crypto_copy_hash_state(struct fcrypt *fcr, uint32_t dst_sid, uint32_t src_sid)
 {
-	struct csession *src_ses, *dst_ses;
+	struct csession *src_ses = NULL;
+	struct csession *dst_ses = NULL;
 	int ret;
 
 	ret = crypto_get_sessions_by_sid(fcr, src_sid, &src_ses,
